airflow/models/dagbag.py

```diff
@@ -450,7 +453,8 @@ def collect_dags(
         # Used to store stats around DagBag processing
         stats = []
 
-        dag_folder = correct_maybe_zipped(dag_folder)
+        # Ensure dag_folder is a str -- it may have been a pathlib.Path
+        dag_folder = correct_maybe_zipped(str(dag_folder))
```

Comment:
Should we also let `correct_maybe_zipped` accept Pathlib too and convert it to str if needed in the function itself ?

Reply:
We could -- I wasn't sure how deep to chase this is all.